AniHyou/Screens/MediaList/ViewModels/MediaListViewModel.swift

```diff
@@ -23,14 +27,31 @@ class MediaListViewModel: ObservableObject {
     @Published var sort: MediaListSort = .updatedTimeDesc
     
     func getUserMediaList() {
-        Network.shared.apollo.fetch(query: UserMediaListQuery(page: .some(currentPage), perPage: .some(25), userId: .some(userId()), type: .some(.case(mediaType)), status: .some(.case(mediaListStatus)), sort: .some([.case(sort), .case(.mediaIdDesc)])), cachePolicy: forceReload ? .fetchIgnoringCacheData : .returnCacheDataElseFetch) { [weak self] result in
+        Network.shared.apollo.fetch(
+            query: UserMediaListQuery(
+                page: .some(currentPage),
+                perPage: .some(25),
+                userId: .some(userId()),
+                type: .some(.case(mediaType)),
+                status: .some(.case(mediaListStatus)),
+                sort: .some([.case(sort), .case(.mediaIdDesc)])
+            ),
+            cachePolicy: forceReload ? .fetchIgnoringCacheData : .returnCacheDataElseFetch) { [weak self] result in
             switch result {
             case .success(let graphQLResult):
                 if let page = graphQLResult.data?.page {
                     if let list = page.mediaList {
+                        list.forEach { self?.mediaSet.insert($0) }
```

Comment:
could also be deleted.. was playing with sets to prevent duplication but it looks like you fixed the root cause @axiel7

Reply:
yes, the duplication was because sometimes the sort option was invalid. For example sorting by score on your planning list, every item had score null so the API returned duplicated results. I solved it by simply sorting also by id.